File: leetcode_py/Reverse Integer.py
class Solution:
    # @return an integer
    def reverse(self, x):
        res=0
        flag=1
        if x<0:
            flag=-1
            x=-x
        while x:
            res*=10
            res+=x%10
            x/=10
            if res >= (1<<32)-1:
                return 0
                
        res*=flag
        return res

File: leetcode_js/Reverse Integer.js
/**
 * @param {number} x
 * @return {number}
 */
var reverse = function(x) {
    var sign = 1;
    if(x < 0){
        sign = -1;
        x = -x;
    }
    var res = 0;
    while(x > 0){
        res *= 10;
        res += x % 10;
        x = Math.floor(x / 10);
        if(res > Math.pow(2, 31) - 1) return 0;
    }
    return res*sign;
};